UIKitDZ/UIKitDZ/AnchorViewController.swift

```diff
@@ -0,0 +1,90 @@
+// AnchorViewController.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// AnchorViewController - страница с реализацией анкеров
+class AnchorViewController: UIViewController {
+    // MARK: Мшуц
+
+    let mainView = UIView()
+    let redLightView = LightView()
+    let yelloyLightView = LightView()
+    let greenLightView = LightView()
+
+    private var mainViewHeightConstraint: NSLayoutConstraint!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+
+    // MARK: - При повороте смартфона
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        coordinator.animate(alongsideTransition: { [weak self] _ in
+            guard let self = self else { return }
+
+            let isPortrait = size.height > size.width
+
+            if isPortrait {
+                self.mainViewHeightConstraint.constant = 376
+            } else {
+                self.mainViewHeightConstraint.constant = 316
+            }
+
+            self.view.layoutIfNeeded()
+        }, completion: nil)
```

Comment:
```suggestion
        coordinator.animate(alongsideTransition: { 
            let isPortrait = size.height > size.width
            self.mainViewHeightConstraint.constant = isPortrait ? 376 : 316
            self.view.layoutIfNeeded()
        }, completion: nil)
```

self не содержит ссылку на coordinator. Следовательно и в списке захвата можно ничего не указывать. И не делать ссылку на self на время работы замыкания.

Вместо if else напрашивается ипользование тернарного оператора.

Reply:
я несколько часов на это ломал голову; думаю что  через якоря нужно делать. Потом нашел такой метод и просто его интергрировал.